frontend/src/lib/components/ImportEmails.svelte

```diff
@@ -0,0 +1,130 @@
+<script lang="ts">
+	import EmailsWarning from '$lib/components/EmailsWarning.svelte';
+	import { FileError } from '$lib/entities/FileError';
+	import { readFile } from '$lib/utils/readFile';
+	import { getErrorMessage } from '$lib/utils/getErrorMessage';
+	import {
+		isErrorModalHidden,
+		errorModalContent,
+		isWarningModalHidden,
+		warningModalContent
+	} from '$lib/stores/global';
+
+	export let users: string[];
+	export let title: string;
+	export let label: string;
+	export let id: string;
+	export let checkKeys: boolean;
+	export let disabled: boolean = false;
+	export let size: number = 1;
+	export let width: string = 'unset';
+
+	let fileElement: HTMLInputElement | null = null;
+	let fileName: string = 'No file selected';
+	let fileWarning: FileError = FileError.NoError;
+
+	async function filterUsers(filter: string, emails: string[]): Promise<string[]> {
+		const response = await fetch(`/api/users/${filter}`, {
+			method: 'POST',
+			body: JSON.stringify({ emails: emails }),
+			headers: {
+				'Content-Type': 'application/json'
+			}
+		});
+
+		if (!response.ok) {
+			const body = await response.json();
+			$errorModalContent = getErrorMessage(body.detail);
+			$isErrorModalHidden = false;
+			return emails;
+		}
+
+		return await response.json();
+	}
+
+	async function handleFileChange() {
+		fileElement = document.querySelector<HTMLInputElement>('#emails-file');
+		fileName = fileElement?.files?.[0]?.name ?? 'No file selected';
+
+		const emails = await processEmails();
+		let invalidEmails: string[] = [];
+
+		if (checkKeys) {
+			invalidEmails = await filterUsers('filter-users-with-no-public-key', emails);
+		} else {
+			invalidEmails = await filterUsers('filter-unregistered-users', emails);
+		}
+
+		if (invalidEmails.length > 0 && $isErrorModalHidden) {
+			$warningModalContent = `Could not import ${invalidEmails.length} users, because they haven't ${checkKeys ? 'generated keys' : 'registered'} yet.`;
+			$isWarningModalHidden = false;
+		}
+
+		const validEmails = emails.filter((e) => !invalidEmails.includes(e));
```

Comment:
we could make invalidEmails a set, then looking up if an email is present in it would have O(1) time complexity instead of being linear